bookkeeper-server/src/test/java/org/apache/bookkeeper/test/TestStatsProvider.java

```diff
@@ -62,12 +64,31 @@ public void dec() {
         @Override
         public void add(long delta) {
             val.addAndGet(delta);
+            updateMax();
         }
 
         @Override
         public Long get() {
             return val.get();
         }
+
+        public void updateMax() {
+            while (true) {
+                long curMax = max.get();
+                long curVal = val.get();
+                if (curVal <= curMax) {
+                    break;
+                } else {
+                    if (max.compareAndSet(curMax, curVal)) {
```

Comment:
I think this can be done better with only one call to accumulateAndGet

Reply:
@eolivelli Ok, how about this version?